### PR TITLE
chore: upgrade `chronos` to address RUSTSEC-2020-0159

### PR DESCRIPTION
**Pull request by diwakergupta**
From `cargo audit`:

```
Crate:     chrono
Version:   0.4.19
Title:     Potential segfault in `localtime_r` invocations
Date:      2020-11-10
ID:        RUSTSEC-2020-0159
URL:       https://rustsec.org/advisories/RUSTSEC-2020-0159
Solution:  Upgrade to >=0.4.20
```

Changelog: https://github.com/chronotope/chrono/releases/tag/v0.4.20

Note that I did not upgrade to the latest dependency -- 0.4.22 -- as it pulls in a few other depenendencies that I wasn't sure about, notably `iana-time-zone`.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -493,14 +493,15 @@ checksum = "baf1de4339761588bc0619e3cbc0120ee582ebb74b53b4efbf79117bd2da40fd"
...
 [[package]]
 name = "chrono"
-version = "0.4.19"
+version = "0.4.20"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "670ad68c9088c2a963aaa298cb369688cf3f9465ce5e2d4ca10e6e0098a1ce73"
+checksum = "6127248204b9aba09a362f6c930ef6a78f2c1b2215f8a7b398c06e1083f17af0"
 dependencies = [
- "libc",
+ "js-sys",
```
Comment (jcnelson) [inline Cargo.lock:500]: Why is `js-sys` a dependency on what should just be a wrapper around `localtime_r`?

Comment (jcnelson) [reply Cargo.lock:500]: Looks like `chronos` by default includes all the dependencies needed to compile to a wasm target.  Let's turn this off:

```
chrono = { version = "0.4.22", features = ["clock"] }
```

Comment (diwakergupta) [reply Cargo.lock:500]: Tried this and doesn't quite work as expected -- 0.4.22 still pulls in js-sys, iana-time-zone as well as android_system_properties.

**Comment by diwakergupta** [comment]
I rebased on top of Aaron's commit to fix the build, and incorporated @jcnelson's suggestion. But looks like `iana-time-zone` also pulls in `js-sys`. Sadly I'm at the limit of my cargo-fu and couldn't figure out how to get rid of the wasm related dependencies. 

If this PR is unacceptable as-is -- someone wants to commandeer this PR, go for it, otherwise feel free to close it out for now.

**Comment by jcnelson** [comment]
I looked into the CVE for this.  Basically, someone needs to be able to race the call to `localtime_r` with a call to `setenv` in a different thread of the process in order to create a dangling pointer which will lead to a segfault.

This is basically unexploitable for us -- we don't set envars on-the-fly in production, and we certainly don't set the `TZ` environment variable anywhere (which is ready by `localtime_r`).

I'm fine with just closing this PR, and maybe documenting this choice somewhere.

**Comment by diwakergupta** [comment]
> I'm fine with just closing this PR, and maybe documenting this choice somewhere.

Thanks for looking into this @jcnelson , closing for now. If someone's motivated enough to address all audit flags and incorporate that into CI, go for it!

**Pull request closed without merging**